src/freenet/node/RequestSender.java

```diff
@@ -1542,9 +1543,9 @@ public synchronized short waitUntilStatusChange(short mask) {
     	}
     }
     
-	private static MedianMeanRunningAverage avgTimeTaken = new MedianMeanRunningAverage();
+	private static final RunningAverage avgTimeTaken = new TrivialRunningAverage();
 	
-	private static MedianMeanRunningAverage avgTimeTakenTransfer = new MedianMeanRunningAverage();
+	private static final RunningAverage avgTimeTakenTransfer = new TrivialRunningAverage();
```

Comment:
Why does this use the TrivialRunningAverage, when there are much more efficient algorithms? This looks like a good case for the now more efficient `BootstrappingDecayingRunningAverage`.

Reply:
MedianMeanRunningAverage is expensive (and is therefore deprecated) as it has O(1) updates but O(n log n) queries and O(n) memory usage.

TrivialRunningAverage is cheap with O(1) updates, O(1) queries and O(1) memory usage.

BootstrappingDecayingRunningAverage might give slightly more useful metrics, but note that `avgTimeTakenTransfer` is only used when `logMINOR` is enabled for this class - this is mostly a debugging tool that probably nobody actually uses.